### PR TITLE
Adding Kotlin examples and one Dart example to 0x05d

### PR DESCRIPTION
**Pull request by B3nac**
Thank you for submitting a Pull Request to the Mobile Security Testing Guide. Please make sure that:

- [ x] Your contribution is written in the 2nd person (e.g. you)
- [ x] Your contribution is written in an active present form for as much as possible.
- [ x] You have made sure that the reference section is up to date (e.g. please add sources you have used, make sure that the references to MITRE/MASVS/etc. are up to date)
- [ x] Your contribution has proper formatted markdown and/or code
- [ x] Any references to website have been formatted as [TEXT](URL “NAME”)
- [ x] You verified/tested the effectiveness of your contribution (e.g.: is the code really an effective remediation? Please verify it works!)

If your PR is related to an issue. Please end your PR test with the following line:
This PR is part of the progress for  #< 962 >.

### CODE REVIEW

**Comment by B3nac** [comment]
Please let me know if any changes are needed. I can add more Dart snippets as I learn how to create apps with Flutter. The plan is to commit Kotlin conversions one page at a time. I was able to debug and convert most of the Java examples on page 0x05d.

File: Document/0x05d-Testing-Data-Storage.md [hunk 1]
```diff
@@ -33,13 +33,34 @@ The following code snippets demonstrate bad practices that disclose sensitive in
...
+var sharedPref = getSharedPreferences("key", Context.MODE_WORLD_READABLE)
+var editor = sharedPref.edit()
+editor.putString("username", "administrator")
+editor.putString("password", "supersecret")
+editor.commit()
+```
+Example for Dart:
+
+```dart
+storeCredentials() async {
```
Comment (commjoen) [inline Document/0x05d-Testing-Data-Storage.md:57]: hi @B3nac , isn't Dart only supported by Flutter? or is it supported by the normal android runtime as well?

**Comment by B3nac** [comment]
Hi @commjoen,

I believe It's built on top of Android runtime. Flutter can actually be
used as a module alongside of a native app programmed in Java/Kotlin.
Vulnerabilities can be introduced via plugins since there's community and
dev versions. I can remove the dart snippit, It's just something newer that
is interesting to research at the moment. :)

**Comment by commjoen** [comment]
Hi Benac, I love the dart content, but for now, maybe it is best to focus on kottlin and java. Did you see  https://github.com/OWASP/owasp-mstg#about-hybrid-apps ? Maybe it is a nice idea to have a look over there and add dart references for Flutter?

**Comment by B3nac** [comment]
Hi @commjoen , no worries I have removed the Dart example and will check out https://github.com/OWASP/owasp-mstg#about-hybrid-apps. I was not aware of this before thank you for sharing.

**Comment by commjoen** [comment]
Ah, i wanted to approve, but our markdown lint check says there are markdown errors. Specifically https://github.com/OWASP/owasp-mstg/pull/1766/checks?check_run_id=736352604 says: 

```
Document/0x05d-Testing-Data-Storage.md: 44: MD031/blanks-around-fences Fenced code blocks should be surrounded by blank lines [Context: "```"]
Document/0x05d-Testing-Data-Storage.md: 279: MD031/blanks-around-fences Fenced code blocks should be surrounded by blank lines [Context: "```"]
Document/0x05d-Testing-Data-Storage.md: 607: MD031/blanks-around-fences Fenced code blocks should be surrounded by blank lines [Context: "```"]
Document/0x05d-Testing-Data-Storage.md: 765: MD031/blanks-around-fences Fenced code blocks should be surrounded by blank lines [Context: "```"]
```

can you fix those please 👍 ?

**Comment by B3nac** [comment]
> Ah, i wanted to approve, but our markdown lint check says there are markdown errors. Specifically https://github.com/OWASP/owasp-mstg/pull/1766/checks?check_run_id=736352604 says:
> 
> ```
> Document/0x05d-Testing-Data-Storage.md: 44: MD031/blanks-around-fences Fenced code blocks should be surrounded by blank lines [Context: "```"]
> Document/0x05d-Testing-Data-Storage.md: 279: MD031/blanks-around-fences Fenced code blocks should be surrounded by blank lines [Context: "```"]
> Document/0x05d-Testing-Data-Storage.md: 607: MD031/blanks-around-fences Fenced code blocks should be surrounded by blank lines [Context: "```"]
> Document/0x05d-Testing-Data-Storage.md: 765: MD031/blanks-around-fences Fenced code blocks should be surrounded by blank lines [Context: "```"]
> ```
> 
> can you fix those please  ?

Hi @commjoen ,

I fixed the space for the code blocks. Looks like the second lint link check failed with `Status: 0 Error: ESOCKETTIMEDOUT`. No new links were added anyway. Onto the next page!

**Comment by commjoen** [comment]
Hi @B3nac ! I don't think we have to fix those at the moment. Having the lint errors fixed is most important. Thanks!

**Pull request merged**